LeanMapperQuery/Entity.php

```diff
@@ -52,7 +54,29 @@ protected function queryProperty($field, IQuery $query)
 		if ($relationship instanceof Relationship\BelongsToMany) {
 			$targetTable = $relationship->getTargetTable();
 			$referencingColumn = $relationship->getColumnReferencingSourceTable();
-			$rows = $this->row->referencing($targetTable, $referencingColumn, new Filtering($filters), $relationship->getStrategy());
+			$strategy = $relationship->getStrategy();
+			$usesDetector = FALSE;
```

Comment:
Navrhuji `$detectStrategy = $strategy !== NULL`. Pak `if ($detectStrategy) { /* add filter */ }`.

Reply:
Upravil jsem to na `$detectStrategy = $strategy !== Result::STRATEGY_UNION`.